module/zfs/dmu_send.c

```diff
@@ -722,6 +844,17 @@ do_dump(dmu_sendarg_t *dsa, struct send_block_record *data)
 		    !split_large_blocks && !BP_SHOULD_BYTESWAP(bp) &&
 		    !BP_IS_EMBEDDED(bp) && !DMU_OT_IS_METADATA(BP_GET_TYPE(bp));
 
+		/*
+		 * Raw sends only apply to protected blocks. Raw sends are
+		 * mutually exclusive with splitting large blocks and
+		 * compressed sends, so we assert that here.
+		 */
+		boolean_t request_raw =
+		    (dsa->dsa_featureflags & DMU_BACKUP_FEATURE_RAW) != 0;
+
+		IMPLY(request_raw, !request_compressed);
```

Comment:
Since raw sends are always compressed, could we treat `request_compressed` as being essentially implied by `request_raw`?  At least, from the CLI it makes sense to do `--compressed --raw` and basically ignore the `--compressed`, instead of treating it as an error (maybe it already does this at a higher layer).